go/engine/team_create.go

```diff
@@ -0,0 +1,334 @@
+	if err != nil {
+		return err
+	}
+
+	perTeamSecret, perTeamSigningKey, perTeamEncryptionKey, err := generatePerTeamKeys()
+	if err != nil {
+		return err
+	}
+
+	ownerLatest := me.GetComputedKeyFamily().GetLatestPerUserKey()
+	if ownerLatest == nil {
+		return errors.New("can't create a new team without having provisioned a per-user key")
+	}
+	secretboxRecipients := map[string]keybase1.PerUserKey{
+		me.GetName(): *ownerLatest,
+	}
+	// These boxes will get posted along with the sig below.
+	secretboxes, err := boxTeamSharedSecret(perTeamSecret, deviceEncryptionKey, secretboxRecipients)
+	if err != nil {
+		return err
+	}
+
+	teamSection, err := makeRootTeamSectionNoReverse(e.name, me, perTeamSigningKey.GetKID(), perTeamEncryptionKey.GetKID())
+	if err != nil {
+		return err
+	}
+
+	// At this point the team section has every field filled out except the
+	// reverse sig. Now we'll wrap it into a full sig, marshal it to JSON, and
+	// sign it, *twice*. The first time with the per-team signing key, to
+	// produce the reverse sig, and the second time with the device signing
+	// key, after the reverse sig has been written in.
+
+	sigBodyForReverse, err := me.TeamRootSig(deviceSigningKey, teamSection)
+	if err != nil {
+		return err
+	}
+	// Note that this (sigchain-v1-style) reverse sig is made with the derived *per-team* signing key.
+	reverseSig, _, _, err := libkb.SignJSON(sigBodyForReverse, perTeamSigningKey)
+	if err != nil {
+		return err
+	}
+	teamSection.PerTeamKey.ReverseSig = &reverseSig
+
+	// Update the team section to include the reverse sig, format it again, and make a sigchain-v2-style sig out of it.
+	sigBodyAfterReverse, err := me.TeamRootSig(deviceSigningKey, teamSection)
+	if err != nil {
+		return err
+	}
+	sigJSONAfterReverse, err := sigBodyAfterReverse.Marshal()
+	if err != nil {
+		return err
+	}
+	v2Sig, err := makeSigchainV2OuterSig(
+		deviceSigningKey,
+		libkb.LinkTypeTeamRoot,
+		1, /* seqno */
+		sigJSONAfterReverse,
+		nil,   /* prevLinkID */
+		false, /* hasRevokes */
+	)
+	if err != nil {
+		return err
+	}
+
+	sigMultiItem := libkb.SigMultiItem{
+		Sig:        v2Sig,
+		SigningKID: deviceSigningKey.GetKID(),
+		Type:       string(libkb.LinkTypeTeamRoot),
+		SigInner:   string(sigJSONAfterReverse),
+		TeamID:     libkb.RootTeamIDFromName(e.name),
+	}
+	sigMultiItem.PublicKeys.Encryption = perTeamEncryptionKey.GetKID()
+	sigMultiItem.PublicKeys.Signing = perTeamSigningKey.GetKID()
+
+	payload := make(libkb.JSONPayload)
+	payload["sigs"] = []interface{}{sigMultiItem}
+	payload["per_team_key"] = secretboxes
+
+	_, err = e.G().API.PostJSON(libkb.APIArg{
+		Endpoint:    "sig/multi",
+		SessionType: libkb.APISessionTypeREQUIRED,
+		JSONPayload: payload,
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (e *TeamCreateEngine) getDeviceKeys(ctx *Context, me *libkb.User) (sigKey libkb.NaclSigningKeyPair, encKey libkb.NaclDHKeyPair, err error) {
+	sigSKA := libkb.SecretKeyArg{
```

Comment:
You don't need this function at all.

e.G().ActiveDevice.SigningKey()
e.G().ActiveDevice.EncryptionKey()